caddyhttp/redirect/setup.go

```diff
@@ -34,10 +34,11 @@ func redirParse(c *caddy.Controller) ([]Rule, error) {
 	cfg := httpserver.GetConfig(c)
 
 	initRule := func(rule *Rule, defaultCode string, args []string) error {
-		if cfg.TLS.Enabled {
-			rule.FromScheme = "https"
-		} else {
-			rule.FromScheme = "http"
+		rule.FromScheme = func() string {
+			if cfg.TLS.Enabled {
+				return "https"
+			}
+			return "http"
```

Comment:
Is this thread-safe?

(If not, and if the solution is to introduce locking, let's reconsider how we do the scheme changing instead...)